src/DelegateDecompiler.Tests/NestedExpressionsTests.cs

```diff
@@ -178,5 +194,107 @@ public void TestFuncWithStaticMethodClosure()
                 ints => ints.SingleOrDefault(i => i == M2())
             );
         }
+
+        [Test, Ignore("Difference is expected")]
+        public void TestQueryableBoundAsVariable()
+        {
+            IQueryable<int> query = Enumerable.Empty<int>().AsQueryable().Where(i => i >= 0);
+            Test<Func<IQueryable<int>, IQueryable<int>>>(
+                ints => Enumerable.Empty<int>().AsQueryable().Where(i => i >= 0),
```

Comment:
I think this test is incorrect. It should be `ints => query`

Reply:
The test is correct when based on the assumption that every nested IQueryable reference encountered should be also decompiled (_I suppose that is the issue you raised by your comment that decompilation may go to far ?_)

The test will fail anyway because of different outputs, but the difference would be acceptable (`System.Int32[]` instead of the expected `Empty().AsQueryable()`)